cloudflare/data_cloudflare_devices.go

```diff
@@ -0,0 +1,73 @@
+package cloudflare
+
+import (
+	"context"
+	"fmt"
+
+	cloudflare "github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceCloudflareDevices() *schema.Resource {
+	return &schema.Resource{
+		Schema: resoureceCloudflareDevicesSchema(),
+		Read:   dataResourceCloudflareDevicesRead,
+		Importer: &schema.ResourceImporter{
+			State: resourceCloudflareDevicesImport,
+		},
+	}
+}
+
+func dataResourceCloudflareDevicesRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+	accountID := d.Get("account_id").(string)
+
```

Comment:
we need to call `d.SetId()` in the `Read` here as nothing else defines the identifier that Terraform references as the "anchor".

```suggestion
	accountID := d.Get("account_id").(string)
	d.SetID(accountID)

```